rules/precompiled_apple_resource_bundle.bzl

```diff
@@ -0,0 +1,235 @@
+"""
+This provides a resource bundle implementation that builds the resource bundle
+only once
+NOTE: This rule only exists because of this issue
+https://github.com/bazelbuild/rules_apple/issues/319
+if this is ever fixed in bazel it should be removed
+"""
+
+load("@bazel_skylib//lib:partial.bzl", "partial")
+load("@bazel_skylib//lib:paths.bzl", "paths")
+load("@build_bazel_rules_apple//apple/internal:apple_product_type.bzl", "apple_product_type")
+load("@build_bazel_rules_apple//apple/internal:intermediates.bzl", "intermediates")
+load("@build_bazel_rules_apple//apple/internal:partials.bzl", "partials")
+load("@build_bazel_rules_apple//apple/internal:resource_actions.bzl", "resource_actions")
+load("@build_bazel_rules_apple//apple/internal:rule_factory.bzl", "rule_factory")
+load("//rules:transition_support.bzl", "transition_support")
+load("@build_bazel_rules_apple//apple:providers.bzl", "AppleResourceBundleInfo", "AppleResourceInfo")
+
+_FAKE_BUNDLE_PRODUCT_TYPE_BY_PLATFORM_TYPE = {
+    "ios": apple_product_type.application,
+    "tvos": apple_product_type.application,
+    "watchos": apple_product_type.watch2_application,
```

Comment:
Should we add `macos` here since _why not_?

Reply:
macos is tested, it just doesnt need a fake bundle product type since the rule descriptors have one for a macos resource bundle